### PR TITLE
[systemd-sonic-generator] replace sscanf with strtol

### PR DESCRIPTION
**Pull request by maipbui**
#### Why I did it
Avoid 'sscanf()' for number conversions. Its use can lead to undefined behavior, slow processing, and integer overflows. Instead prefer the 'strto*()' family of functions.
##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
replace sscanf with strtol
#### How to verify it
Manual test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/systemd-sonic-generator/systemd-sonic-generator.c [hunk 1]
```diff
@@ -569,7 +569,11 @@ int get_num_of_asic() {
                     str_num_asic = strtok_r(NULL, "=", &saveptr);
                     strip_trailing_newline(str_num_asic);
                     if (str_num_asic != NULL){
-                        sscanf(str_num_asic, "%d",&num_asic);
+                        char *endptr;
+                        num_asic = strtol(str_num_asic, &endptr, 10);
+                        if (*endptr != '\0') {
+                            fprintf(stderr, "Failed to get num asics.\n");
```
Comment (qiluo-msft) [inline src/systemd-sonic-generator/systemd-sonic-generator.c:575]: >fprintf [](http://example.com/codeflow?start=28&length=7)

You will treat any extra chars after num as failure. I am thinking original sscanf just ignore them, and it's safe to ignore them using strtol.

How about just checking `num_asic > 0` ?

**Review by qiluo-msft — APPROVED** [review]
LGTM

File: src/systemd-sonic-generator/systemd-sonic-generator.c [hunk 2]
```diff
@@ -569,9 +569,11 @@ int get_num_of_asic() {
                     str_num_asic = strtok_r(NULL, "=", &saveptr);
                     strip_trailing_newline(str_num_asic);
                     if (str_num_asic != NULL){
-                        sscanf(str_num_asic, "%d",&num_asic);
+                        num_asic = strtol(str_num_asic, NULL, 10);
+                        if (num_asic > 0) {
+                            break;
```
Comment (SuvarnaMeenakshi) [inline src/systemd-sonic-generator/systemd-sonic-generator.c:574]: I did not understand the usage of break here?
If num_asic is not greater than 0, then what might happen?

In what scenario will this check be useful? 
Previously, we were breaking after parsing the line starting with "NUM_ASIC=", we could still do the same without check for the value?

**Review by SuvarnaMeenakshi — APPROVED** [review]
lgtm

**Pull request merged**